File: app/src/main/java/pro/xite/dev/vkgram/localalbum/LocalPicturesAlbumFragment.java
package pro.xite.dev.vkgram.localalbum;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.arellomobile.mvp.MvpAppCompatFragment;
import com.arellomobile.mvp.presenter.InjectPresenter;
import com.arellomobile.mvp.presenter.ProvidePresenter;

import butterknife.BindView;
import butterknife.ButterKnife;
import pro.xite.dev.vkgram.R;
import timber.log.Timber;

public class LocalPicturesAlbumFragment extends MvpAppCompatFragment implements AlbumView {

    @InjectPresenter
    LocalAlbumPresenter p;

    @BindView(R.id.recycler_view) RecyclerView recyclerView;

    public LocalPicturesAlbumFragment() {}

    public static LocalPicturesAlbumFragment newInstance() {
        return new LocalPicturesAlbumFragment();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        Timber.d("onCreateView: ");
        return inflater.inflate(R.layout.content_recycler, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        Timber.d("onViewCreated");
        super.onViewCreated(view, savedInstanceState);
        ButterKnife.bind(this, view);
        loadPictures();
    }

    @ProvidePresenter
    public LocalAlbumPresenter providePresenter() {
        Timber.d("providePresenter");
        return new LocalAlbumPresenter();
    }

    private void loadPictures() {

    }

    @Override
    public void initRecyclerView() {
        Timber.d("initRecycler: ");
        recyclerView.setLayoutManager(new GridLayoutManager(getContext(),2));
        recyclerView.setAdapter(new LocalAlbumAdapter());
    }
}

File: app/src/main/java/pro/xite/dev/vkgram/localalbum/LocalPicturesAlbumFragment.kt
package pro.xite.dev.vkgram.localalbum

import android.os.Bundle
import android.support.v7.widget.GridLayoutManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.arellomobile.mvp.MvpAppCompatFragment
import com.arellomobile.mvp.presenter.InjectPresenter
import com.arellomobile.mvp.presenter.ProvidePresenter
import kotlinx.android.synthetic.main.content_recycler.*
import pro.xite.dev.vkgram.R
/**
 * Created by Roman Syrchin on 6/3/18.
 */
class LocalPicturesAlbumFragment : MvpAppCompatFragment(), AlbumView  {

    @InjectPresenter
    lateinit var p : LocalAlbumPresenter

    companion object {
        @JvmStatic
        fun create(): LocalPicturesAlbumFragment = LocalPicturesAlbumFragment ()
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.content_recycler, container, false)
    }

    @ProvidePresenter
    fun provideAlbumPresent() : LocalAlbumPresenter {
        return LocalAlbumPresenter(LocalAlbumModel(context))
    }

    override fun initRecyclerView() {
        recycler_view.setItemViewCacheSize(22)
        recycler_view.layoutManager = GridLayoutManager(context, 2)
        recycler_view.adapter = LocalAlbumAdapter(p)
    }


}

